### ISSUE TITLE
[Issue]: Creating a new library and then immediately renaming it causes a ghost library with no folder located in config/root, Ghost library cannot be deleted, Scan All Media fails

### ISSUE THREAD

**Issue by gummykage** [jellyfin/jellyfin#11885]
### Please describe your bug

While trying to figure out why my 10.8.13 to 10.9.3 is having so many issues with freezing interface on ios and safari, I created a new server on win10 (Latest updates) and started adding libraries to see how it handles it. I created a library and noticed that it was named wrong, so immediately renamed it. Upon Full media scan a ghost of the original library named wrong appears on the home screen with no folder representing it along with the correctly named library. The ghost library also links to the renamed library... Now Scan All Libraries constantly fails.

### Reproduction Steps

Create library. Immediately rename it (library had movies in it). Ghost of original name library along side the one named correctly will appear. Scan All Media will not fail.

![image](https://github.com/jellyfin/jellyfin/assets/9695713/93d2ac50-9fdb-4e8f-92d9-b7608564b8bd)

![image](https://github.com/jellyfin/jellyfin/assets/9695713/78f1ea60-d629-4e14-9792-daa17f2d4b22)

![image](https://github.com/jellyfin/jellyfin/assets/9695713/a1c1e222-49b5-4f91-bcae-4ebc4d168969)

### Jellyfin Version

10.9.0

### if other:

10.9.3

### Environment

```markdown
- OS:Win10
- Linux Kernel:
- Virtualization:
- Clients: Jellyfin Media Player latest
- Browser:
- FFmpeg Version:
- Playback Method:
- Hardware Acceleration:
- GPU Model:
- Plugins:
- Reverse Proxy:
- Base URL:
- Networking:
- Storage:
```

### Jellyfin logs

```shell
[11:53:07] [ERR] [24] Emby.Server.Implementations.ScheduledTasks.TaskManager: Error executing Scheduled Task
System.IO.DirectoryNotFoundException: Could not find a part of the path 'H:\ProgramData\Jellyfin\Server\root\default\Mov
ies2'.
   at System.IO.Enumeration.FileSystemEnumerator`1.CreateDirectoryHandle(String path, Boolean ignoreNotFound)
   at System.IO.Enumeration.FileSystemEnumerator`1.Init()
   at System.IO.Enumeration.FileSystemEnumerable`1..ctor(String directory, FindTransform transform, EnumerationOptions o
ptions, Boolean isNormalized)
   at System.IO.Enumeration.FileSystemEnumerableFactory.FileSystemInfos(String directory, String expression, Enumeration
Options options, Boolean isNormalized)
   at System.IO.DirectoryInfo.InternalEnumerateInfos(String path, String searchPattern, SearchTarget searchTarget, Enume
rationOptions options)
   at System.IO.DirectoryInfo.EnumerateFileSystemInfos(String searchPattern, EnumerationOptions enumerationOptions)
   at Emby.Server.Implementations.IO.ManagedFileSystem.GetFileSystemEntries(String path, Boolean recursive)
   at MediaBrowser.Controller.Providers.DirectoryService.<>c.<GetFileSystemEntries>b__5_0(String p, IFileSystem fileSyst
em)
   at System.Collections.Concurrent.ConcurrentDictionary`2.GetOrAdd[TArg](TKey key, Func`3 valueFactory, TArg factoryArg
ument)
   at MediaBrowser.Controller.Providers.DirectoryService.GetFileSystemEntries(String path)
   at MediaBrowser.Controller.IO.FileData.GetFilteredFileSystemEntries(IDirectoryService directoryService, String path,
IFileSystem fileSystem, IServerApplicationHost appHost, ILogger logger, ItemResolveArgs args, Int32 flattenFolderDepth,
Boolean resolveShortcuts)
   at MediaBrowser.Controller.Entities.CollectionFolder.CreateResolveArgs(IDirectoryService directoryService, Boolean se
tPhysicalLocations)
   at MediaBrowser.Controller.Entities.CollectionFolder.RequiresRefresh()
   at MediaBrowser.Providers.Manager.MetadataService`2.RefreshMetadata(BaseItem item, MetadataRefreshOptions refreshOpti
ons, CancellationToken cancellationToken)
   at MediaBrowser.Controller.Entities.BaseItem.RefreshMetadata(MetadataRefreshOptions options, CancellationToken cancel
lationToken)
   at Emby.Server.Implementations.Library.LibraryManager.ValidateTopLibraryFolders(CancellationToken cancellationToken,
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by gummykage** [comment]
Work around, rename the library back to the original name, delete it, then recreate it.

**Comment by lukebiddell** [comment]
Just encountered the same issue. Did deleting the ghost library not delete your media files?

![Screenshot_20240531-015906.png](https://github.com/jellyfin/jellyfin/assets/18555718/4a93c182-8506-4e8d-b36e-811c4b806bb7)

**Comment by gummykage** [comment]
No it just deleted the library reference. It's the same as removing it from the dashboard.

**Comment by gummykage** [comment]
I love 10.9.3's new features so much, but some bugs are just too intense that I found myself every night trying to find work arounds for so many new bugs.

I've reverted back to 10.8.13 on docker. Remember to add back "DOTNET_SYSTEM_GLOBALIZATION_INVARIANT=1"
ios jellyfin mobile app - still using to old one from a year ago no longer has the library carousel freezing problem anymore
jellyfin-androidtv - Fire Stick TV using adb reverted the app back to jellyfin-androidtv-v0.16.10-release.apk no longer have that mandatory to update to 10.9.x message anymore
Jellyfin Media Player win10 reverted back to 1.9.1

On win10 I have 10.9.3 installed as a test server to find bugs.
Ghost libraries was kind of a tipping point where I realized so many QOL things we were enjoying is currently broken. I feel bad I didn't something when I was testing the unstable branch.

**Comment by gnattu** [comment]
A fix is on the way and will contain mitigation method for you to remove the ghosting library

**Comment by seascape** [comment]
May I ask what the built-in mitigation is? I just experienced the ghost library bug on 10.9.3, updated to 10.9.6, and still have the ghost library.

I can do the SQL db edit suggested above if need be, but I assume that is not the mitigation you had in mind.

**Comment by felix920506** [comment]
Should be fixed by #11963
